lgc/patch/PatchInOutImportExport.cpp

```diff
@@ -2615,6 +2617,48 @@ Value *PatchInOutImportExport::patchFsBuiltInInputImport(Type *inputTy, unsigned
       input = builder.CreateInsertElement(input, kCoord, uint64_t(1));
       break;
     }
+    case PrimitiveType::Triangle_Strip:
+    case PrimitiveType::Triangle_Fan:
+    case PrimitiveType::Triangle_Strip_Adjacency: {
+      // Odd
+      auto kCoord = builder.CreateFSub(ConstantFP::get(Type::getFloatTy(*m_context), 1.0), iCoord);
+      kCoord = builder.CreateFSub(kCoord, jCoord);
+      Value *odd = UndefValue::get(inputTy);
+      odd = builder.CreateInsertElement(odd, iCoord, uint64_t(2));
+      odd = builder.CreateInsertElement(odd, jCoord, uint64_t(0));
+      odd = builder.CreateInsertElement(odd, kCoord, uint64_t(1));
+      // Even
+      Value *even = UndefValue::get(inputTy);
+      even = builder.CreateInsertElement(even, iCoord, uint64_t(1));
+      even = builder.CreateInsertElement(even, jCoord, uint64_t(2));
+      even = builder.CreateInsertElement(even, kCoord, uint64_t(0));
+
+      auto primitiveId = patchFsBuiltInInputImport(builder.getInt32Ty(), BuiltInPrimitiveId, nullptr, insertPos);
+      auto evenV = builder.CreateSRem(primitiveId, builder.getInt32(2));
+      Value *con = builder.CreateICmpEQ(evenV, builder.getInt32(0));
+      input = builder.CreateSelect(con, even, odd);
+      break;
+    }
+    case PrimitiveType::Triangle_List_Adjacency: {
+      // Odd
+      auto kCoord = builder.CreateFSub(ConstantFP::get(Type::getFloatTy(*m_context), 1.0), iCoord);
```

Comment:
`Type::getFloatTy(*m_context)` could be `builder.getFloatTy()` to be consistent with the `getInt32Ty`s.